### DISCUSSION TITLE
非開発者の書き込みと、自作スクリプトの紹介について

### DISCUSSION THREAD

**Discussion by DesertDoggy** [increments/qiita-discussions#841] [ヘルプ]
上記について質問があり、こちらに誘導されたので書き込ませて頂きます。
背景を申し上げますと当方、開発者ではありませんが、最近私的な用途で満足の行くソフトがなかったために、スクリプトを書き始めある程度形になったので一般の方で他にもお困りの人がいれば使っていただきたいと作ってあったアカウントで調べもので以前より何かと参考にさせていただいていたQiitaに書き込もうと思ったところエンジニア用の情報交換サイトであるということを恥ずかしながらはじめて気づき、タイトルの質問が生じた次第です。まず一つは職業エンジニアではないのでそもそも書き込む資格があるのかどうかということと(こちらはプログラミング勉強中でもカウントされるとのことと拝見したので大丈夫かと思いますが)、スクリプトの紹介となると商用ではありませんが宣伝に該当してしまう可能性と、プログラミング自体には関連性はあるとはいえエンジニアの方の参考になる情報が提供できるかといわれると全く自信がないため僭越ながら確認させて頂ければと存じます。

**Comment by tanaka-asami415** [comment]
@DesertDoggy 
Qiitaをご検討いただきありがとうございます！

今後、DesertDoggyさんと同じ状況の方など、同様の問題で困ったりした時に記事があると助かると思いますので、ぜひご投稿をお待ちしております✨

宣伝行為に該当するかどうかについては、ユーザーさまからの通報も判断基準となっていますが
基本的には、Qiita利用規約（ https://qiita.com/terms ）に記載の「客観的には事実上」広告・宣伝行為となっていない状態（記事の内容に対して、アプリやサービス紹介等の部分が主目的に見えない状態）であれば問題ございません。

> Qiita利用規約（第11条第5項第4号）
「客観的には事実上、広告・宣伝や商用を目的とした勧誘と認められる行為（検索サイト最適化又はアフィリエイトを目的とする投稿行為を含むがこれに限らない）。ただし、当社が本サービス上で別途定める場合はその限りではありません。」

目安としては、技術的な解説を主目的にしていただいた上で、作成したアプリ等のご紹介等が一部入っていている状態（宣伝が1~2割程度）であれば、問題ないかと存じます。

ただし、ユーザーさまからの通報も判断基準となっておりますので
通報が一定数に達しましたら、弊社から記事の著者に記事を修正していただきたいという旨をお知らせします。
その際は記事の修正を行っていただければ幸いです。

よろしくお願いします！

**Reply by DesertDoggy** [reply]
早速のご返信とご配慮頂きありがとうございます。修正の必要な場合について承知いたしました。
参考になるかは兎も角、途中行き詰まって処理を工夫した点などあるのでそのあたりを中心に書いて行こうと思います。
重ねて感謝いたします。
